contracts/src/tokens/river/base/delegation/MainnetDelegation.sol

```diff
@@ -47,6 +47,25 @@ contract MainnetDelegation is
     _;
   }
 
+  // =============================================================
+  //                           Batch Delegation
+  // =============================================================
+  function setBatchDelegation(
+    address[] calldata delegators,
+    address[] calldata delegates,
+    address[] calldata claimers,
+    uint256[] calldata quantities
+  ) external onlyCrossDomainMessenger {
+    for (uint256 i = 0; i < delegators.length; i++) {
+      _replaceDelegation(
```

Comment:
What would happen if the arrays are not of equal length ? Maybe we should check and revert prior to entering the for loop if the array sizes differ from one another.